woocommerce/woocommerce
# [Enhancement]: Add option to show price on the place order button

### Changes proposed in this Pull Request:

This PR contains a few changes to get this feature working and some tech-debt improvements:

- Move the non-dynamic attributes to `block.json`
- Add an attribute called `showPrice` - used to control whether the price is shown on the button in checkout
- Add an Inspector Control to toggle this attribute on or off.
- Update the `EditableButton` component to accept children. Required to show the separator and price alongside the text.
- Update the `PlaceOrderButton` component to show the price and separator if `showPrice` is true.
- Add CSS to control the display of the text, separator, and button.
- Update the `PlaceOrderButton` component to add `Spinner` as a child rather than using the deprecated `showSpinner` prop.
- Keep the text inside the `PlaceOrderButton` while it is processing. Previously this was removed during processing and replaced with the spinner, but if the button is taller than the spinner, there will be a layout shift. Keeping the text, but hiding it with CSS prevents the layout shift.





Closes #48433.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add an item to your cart and go to the Checkout page.
2. Look at the place order button. It should not have a price or separator visible.
3. Press the button to place your order. Ensure the spinner shows, and then disappears when the checkout is done processing.
4. Go to the Checkout page in the post editor.
5. Select the Checkout Actions Block (The place order button).
6. In the sidebar, toggle the `Show price in the button` option **on**.
7. In the editor, look at the place order button. Ensure the price is visible with a separator between the text and the price.
9. Ensure you can edit the text (click into it and type, it's not super obvious it's editable).
10. Try with different lengths of text and ensure the button behaves reasonably, i.e. doesn't grow/shrink too much, content doesn't overflow, the price and text have enough room etc.
11. Save your changes to the text and visit the front end. Repeat steps 1, 2, and 3 ensuring your new text is visible, and that the place order button is still working normally.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
